Unfortunately quick-error does not have a Changelog, but it appears that
"description" was made into a no-op in version 1.2.3 and removed completely
in version 2.0.0.

This patch replaces "description" with "display" where there is not already
a "display" implementation and comments out "description" if there is already
a display implementation.

Index: dns-parser/Cargo.toml
===================================================================
--- dns-parser.orig/Cargo.toml
+++ dns-parser/Cargo.toml
@@ -25,7 +25,7 @@ license = "MIT/Apache-2.0"
 version = "1"
 
 [dependencies.quick-error]
-version = "1.0.0"
+version = "2.0.0"
 
 [dependencies.serde]
 version = "1.0"
Index: dns-parser/src/error.rs
===================================================================
--- dns-parser.orig/src/error.rs
+++ dns-parser/src/error.rs
@@ -7,65 +7,65 @@ quick_error! {
         /// Invalid compression pointer not pointing backwards
         /// when parsing label
         BadPointer {
-            description("invalid compression pointer not pointing backwards \
+            display("invalid compression pointer not pointing backwards \
                          when parsing label")
         }
         /// Packet is smaller than header size
         HeaderTooShort {
-            description("packet is smaller than header size")
+            display("packet is smaller than header size")
         }
         /// Packet ihas incomplete data
         UnexpectedEOF {
-            description("packet is has incomplete data")
+            display("packet is has incomplete data")
         }
         /// Wrong (too short or too long) size of RDATA
         WrongRdataLength {
-            description("wrong (too short or too long) size of RDATA")
+            display("wrong (too short or too long) size of RDATA")
         }
         /// Packet has non-zero reserved bits
         ReservedBitsAreNonZero {
-            description("packet has non-zero reserved bits")
+            display("packet has non-zero reserved bits")
         }
         /// Label in domain name has unknown label format
         UnknownLabelFormat {
-            description("label in domain name has unknown label format")
+            display("label in domain name has unknown label format")
         }
         /// Query type code is invalid
         InvalidQueryType(code: u16) {
-            description("query type code is invalid")
+            //description("query type code is invalid")
             display("query type {} is invalid", code)
         }
         /// Query class code is invalid
         InvalidQueryClass(code: u16) {
-            description("query class code is invalid")
+            //description("query class code is invalid")
             display("query class {} is invalid", code)
         }
         /// Type code is invalid
         InvalidType(code: u16) {
-            description("type code is invalid")
+            //description("type code is invalid")
             display("type {} is invalid", code)
         }
         /// Class code is invalid
         InvalidClass(code: u16) {
-            description("class code is invalid")
+            //description("class code is invalid")
             display("class {} is invalid", code)
         }
         /// Invalid characters encountered while reading label
         LabelIsNotAscii {
-            description("invalid characters encountered while reading label")
+            display("invalid characters encountered while reading label")
         }
         /// Invalid characters encountered while reading TXT
         TxtDataIsNotUTF8(error: Utf8Error) {
-            description("invalid characters encountered while reading TXT")
+            //description("invalid characters encountered while reading TXT")
             display("{:?}", error)
         }
         /// Parser is in the wrong state
         WrongState {
-            description("parser is in the wrong state")
+            display("parser is in the wrong state")
         }
         /// Additional OPT record found
         AdditionalOPT {
-            description("additional OPT record found")
+            display("additional OPT record found")
         }
     }
 }
